flink-tests/src/test/java/org/apache/flink/test/state/TaskManagerWideRocksDbMemorySharingITCase.java

```diff
@@ -0,0 +1,257 @@
+    private static final double WRITE_BUFFER_RATIO = 0.5;
+    private static final double EXPECTED_BLOCK_CACHE_SIZE =
+            calculateActualCacheCapacity(
+                    (long) (MANAGED_MEMORY_SIZE_BYTES * MANAGED_MEMORY_SHARED_FRACTION),
+                    WRITE_BUFFER_RATIO);
+    // try to check that the memory usage is limited
+    // however, there is no hard limit actually
+    // because of https://issues.apache.org/jira/browse/FLINK-15532
+    private static final double EFFECTIVE_LIMIT = EXPECTED_BLOCK_CACHE_SIZE * 1.25;
+
+    private InMemoryReporter metricsReporter;
+    private MiniClusterWithClientResource cluster;
+
+    @Before
+    public void init() throws Exception {
+        metricsReporter = InMemoryReporter.create();
+        cluster =
+                new MiniClusterWithClientResource(
+                        new MiniClusterResourceConfiguration.Builder()
+                                .setConfiguration(getConfiguration(metricsReporter))
+                                .setNumberTaskManagers(1)
+                                .setNumberSlotsPerTaskManager(NUMBER_OF_TASKS)
+                                .build());
+        cluster.before();
+    }
+
+    @After
+    public void destroy() {
+        cluster.after();
+        metricsReporter.close();
+    }
+
+    @Test
+    public void testBlockCache() throws Exception {
+        List<JobID> jobIDs = new ArrayList<>(NUMBER_OF_JOBS);
+        try {
+            // launch jobs
+            for (int i = 0; i < NUMBER_OF_JOBS; i++) {
+                jobIDs.add(cluster.getRestClusterClient().submitJob(dag()).get());
+            }
+
+            // wait for init
+            Deadline initDeadline = Deadline.fromNow(Duration.ofMinutes(1));
+            for (JobID jid : jobIDs) {
+                waitForAllTaskRunning(cluster.getMiniCluster(), jid, false);
+                waitForAllMetricsReported(jid, initDeadline);
+            }
+
+            // check declared capacity
+            collectGaugeValues(jobIDs, "rocksdb.block-cache-capacity")
+                    .forEach(
+                            size ->
+                                    assertEquals(
+                                            "Unexpected rocksdb block cache capacity",
+                                            EXPECTED_BLOCK_CACHE_SIZE,
+                                            size,
+                                            0));
+
+            // do some work and check the actual usage of memory
+            for (int i = 0; i < 10; i++) {
+                Thread.sleep(50L);
+                DoubleSummaryStatistics stats =
+                        collectGaugeValues(jobIDs, "rocksdb.block-cache-usage")
+                                .collect(Collectors.summarizingDouble((Double::doubleValue)));
+                assertEquals(
+                        String.format(
+                                "Block cache usage reported by different tasks varies too much: %s\n"
```

Comment:
What's the idea behind verifying that the shared memory metric doesn't change much?  If you were accidentally allocating from exclusive memory, wouldn't you expect that this doesn't deviate much as well?  Or does this measure the cache used regardless of whether it's exclusive memory or not?

Reply:
The idea is that if all rocksdb instances are backed by the same block cache / write buffer manager,
then the metric must show exactly the same usage (modulo reporting delay and measurement lags).

If each rocksdb instance uses its own cache, then the usages will likely be different because of different state growth rates in each subtask.

This is more of a heuristic, but I couldn't come up with a better way to check that there is only a single BlockCache/WBM behind the scenes.

Reply:
Actually, shared memory metrics seem a good idea in general. 
But adding them isn't super trivial (again configuration and passing the metrics should be thought through);
so I'd do it separately, maybe after some feedback for this feature.
And testing directly via task-level metrics seems a bit more robust.
WDYT?

There is no way to access RocksDB internal state at this level AFAIK.